### PR TITLE
Reverse execution order in Auditcare Migration

### PR DESCRIPTION
**Pull request by AmitPhulera**
## Summary

A part of Auditcare Migration - 
https://dimagi-dev.atlassian.net/browse/SAAS-12241

This builds on top of https://dimagi-dev.atlassian.net/browse/SAAS-12241 and the base has been set on that branch itself. 

The changes alter the execution order which was earlier from older records to recent records. These changes enable the migration process to run from the latest records to older records.
## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Automated test coverage

There are a fair amount of tests that were added/updated to ensure that the changes work fine.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

**Comment by AmitPhulera** [comment]
I think I need to rebase it properly. Currently, the base is set to `master` which should be `ap/auditcare-migration`. I would rebase before changing the base.

**Comment by AmitPhulera** [comment]
> I think I need to rebase it properly. Currently, the base is set to `master` which should be `ap/auditcare-migration`. I would rebase before changing the base.

This is done.

File: corehq/apps/auditcare/utils/migration.py [hunk 1]
```diff
@@ -22,30 +22,33 @@ def get_next_batch_start(self):
 
     def generate_batches(self, worker_count, batch_by):
         batches = []
-        # todo : Change get_fixed_start_date_for_sql to something generic
-        cutoff_time = get_fixed_start_date_for_sql()
-        if not cutoff_time:
-            cutoff_time = datetime.now()
         with cache.lock(self.start_lock_key, timeout=10):
             start_datetime = self.get_next_batch_start()
+            print(start_datetime)
```
Comment (millerdev) [inline corehq/apps/auditcare/utils/migration.py:27]: Not sure why I didn't catch this before, but print statements are not allowed except in management commands. I realize this utility function is only used by a management command, so it's a bit of a gray area. Is this even needed except for debugging? If yes, consider using `logging` instead.

File: corehq/apps/auditcare/utils/migration.py [hunk 2]
```diff
@@ -22,30 +22,33 @@ def get_next_batch_start(self):
...
+                    You can manually set key using AuditCareMigraionUtil.set_next_batch_start()""")
+                # For first run set the start_datetime to the event_time of the first record
+                # in the SQL. If there are no records in SQL, start_time would be set as
+                # current time
+                start_datetime = get_sql_start_date()
+                if not start_datetime:
+                    start_datetime = datetime.now()
+
+            if start_datetime < CUTOFF_TIME:
                 print("Migration Successfull")
```
Comment (millerdev) [inline corehq/apps/auditcare/utils/migration.py:41]: print!

**Review by millerdev — APPROVED** [review]

**Comment by AmitPhulera** [comment]
I am merging this PR to `ap/auditcare-migration` and would be addressing the feedback there.

**Pull request merged**